Okta/OktaLogout.swift

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2017, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+import Hydra
+
+public struct Logout {
+    private let idToken: String?
+    
+    init(idToken: String?) {
+        self.idToken = idToken
+    }
+    
+    public func start(withPListConfig plistName: String?, view: UIViewController) -> Promise<Void> {
+        return Promise<Void>(in: .background, { resolve, reject, _ in
+            guard let plist = plistName,
+                  let config = Utils.getPlistConfiguration(forResourceName: plist) else {
+                return reject(OktaError.NoPListGiven)
+            }
+
+            OktaAuthorization().logout(config, idToken: self.idToken, view: view)
+                .then { _ in return resolve() }
+                .catch { error in return reject(error) }
+        })
+    }
+    
+    public func start(_ view: UIViewController) -> Promise<Void> {
```

Comment:
Lets add another `start` method that accepts a `[String:String]` object, similar to [`Login`](https://github.com/okta/okta-sdk-appauth-ios/blob/master/Okta/OktaLogin.swift#L33-L36).

Reply:
Done